src/Pages/Profile/admin/AdminView.jsx

```diff
@@ -0,0 +1,294 @@
+        window.alert('The Email has already existed, leave it blank if you do not want to change')
+        console.log(err)
+      })
+  }
+
+  function editModalButton() {
+    let FormGroups = [
+      {
+        label: 'First Name',
+        type: 'email',
+        placeholder: user.firstName,
+        onChange: (e) => setFirstName(e.target.value)
+      },
+      {
+        label: 'Last Name',
+        type: 'email',
+        placeholder: user.lastName,
+        onChange: (e) => setLastName(e.target.value)
+      },
+      {
+        label: 'Middle Initial',
+        type: 'email',
+        placeholder: user.middleInitial,
+        onChange: (e) => setMiddleInitial(e.target.value)
+      },
+      {
+        label: 'Email',
+        type: 'email',
+        placeholder: user.email,
+        onChange: (e) => setEmail(e.target.value)
+      },
+      {
+        label: 'Password',
+        type: 'password',
+        placeholder: 'make it secure',
+        onChange: (e) => setPassword(e.target.value)
+      },
+      {
+        label: 'Door Code',
+        type: 'email',
+        placeholder: 'make it secure',
+        onChange: (e) => setDoorCode(e.target.value)
+      },
+      {
+        label: 'Major',
+        type: 'email',
+        placeholder: user.major,
+        onChange: (e) => setMajor(e.target.value)
+      }
+    ]
+
+    return (
+      <div>
+        <div>
+          <Button
+            color="primary"
+            style={{
+              position: 'relative',
+              left: '80%'
+            }}
+            onClick={() => { handleToggle() }}>
+            Edit
+          </Button>
+        </div>
+
+        <Modal isOpen={toggle}>
+          <ModalHeader>Modal title</ModalHeader>
+          <ModalBody>
+            <Form>
+
+              {FormGroups.map((group, index) => {
+                return <FormGroup key={index}>
+                  <Label>{group.label}</Label>
+                  <Input
+                    type={group.type}
+                    name={group.type}
+                    placeholder={group.placeholder}
+                    onChange={group.onChange}
+                  />
+                </FormGroup>
+              })}
+
+              Change validation date to
+              <select onChange={(e)=>{setNumberOfSemestersToSignUpFor(e.target.value)}}>
+                <option value={0}>Keep Same</option>
+                <option value={1}>This semester</option>
+                <option value={2}>2 semesters</option>
+              </select>
+
+              <Button
+                type='button'
+                onClick={() => {
+                  setPagesPrinted(0);
+                }}
+                color="info"
+                style={{marginTop:'5px'}}>
+                Reset Pages!
+              </Button>
+
+              <FormGroup tag="fieldset">
+                <legend>Membership Status</legend>
+                {enums.getAllValues(enums.membershipStatus).map((membership, index) => {
```

Comment:
This `return` block should be it's own react component. i dont wanna merge things that may be hard to maintain later. see #61